src/Pages/CardReader/CardReader.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from 'react';
+import { BASE_API_URL } from '../../Enums';
+
+export default function CardReader(props) {
+  const [logs, setLogs] = useState([]);
+  const [error, setError] = useState('');
+  const token = props.user.token;
+
+  const buildLog = (data) => {
+    let endpoint = data.endpoint;
+    if (!endpoint.includes('?add=1')) {
+      endpoint += '      ';
+    }
+    return [new Date().toISOString(), endpoint, data.statusCode, data.message].join('        ');
+  };
+
+  useEffect(() => {
+    const url = new URL('/api/OfficeAccessCard/listen', BASE_API_URL);
+    url.searchParams.append('token', token);
+    const eventSource = new EventSource(url.href);
+    eventSource.onmessage = (event) => {
+      let data = JSON.parse(event.data);
+      let newLog = buildLog(data);
+      setLogs(currLogs => [newLog, ...currLogs]); // prepend the new log
```

Comment:
lets put a try/catch in here, so if for some reason the data cant be parsed we do

```js
try {
      let data = JSON.parse(event.data);
      let newLog = buildLog(data);
      setLogs(currLogs => [newLog, ...currLogs]);
} catch (e) {
      console.error('unable to format event data from /listen', e);
      setLogs(
            (currLogs) => [
                  '[error] unable to format response, check browser logs',
                  ...currLogs,
            ]
      );
}
```

Reply:
also the above code likely needs to be linted